src/server.es6.js

```diff
@@ -50,14 +71,36 @@ class ServerReactApp extends App {
         yield app.route(this);
       }
 
-      if (typeof this.body === 'object' && React.isValidElement(this.body)) {
+      if (typeof this.body === 'function') {
+        // Load all the data required for the request before the server renders
+        var data;
+        this.props = this.props || {};
+
+        try {
+          data = yield app.loadData;
+        } catch (e) {
+          app.error(e, this, app);
+        }
+
+        this.props.dataCache = {};
+
+        if (data) {
+          // The entries are in the same order as when we fired off the promises;
+          // load the data from the response array.
+          var i = 0;
+          for (var [key, value] of this.props.data.entries()) {
+            this.props.dataCache[key] = data[i];
+            i++;
+          }
+        }
+
         yield app.render;
 
         if (formatProps) {
           this.props = formatProps(this.props);
         }
 
-        yield app.injectBootstrap;
+        yield app.injectBootstrap(app.config.formatBootstrap);
```

Comment:
A function would allow you to inject additional clientside-only config as well as whitelisting.